packages/material-ui-styles/src/makeStyles.js

```diff
@@ -194,25 +193,22 @@ function makeStyles(stylesOrCreator, options = {}) {
     name,
     // Help with debuggability.
     classNamePrefix: classNamePrefixOption,
-    defaultTheme: defaultThemeOption,
     Component,
+    defaultTheme = noopTheme,
     ...stylesOptions2
   } = options;
   const stylesCreator = getStylesCreator(stylesOrCreator);
-  const listenToTheme = stylesCreator.themingEnabled || typeof name === 'string';
-  const defaultTheme = defaultThemeOption || noopTheme;
-
   const classNamePrefix = name || classNamePrefixOption || 'Hook';
-
   stylesCreator.options = {
     index: increment(),
     name,
     meta: classNamePrefix,
     classNamePrefix,
   };
+  const listenToTheme = stylesCreator.themingEnabled || typeof name === 'string';
 
   return (props = {}) => {
-    const theme = listenToTheme ? React.useContext(ThemeContext) || defaultTheme : defaultTheme;
+    const theme = (listenToTheme ? useTheme() : null) || defaultTheme;
```

Comment:
I would go with the pattern suggested from the react issue and call a custom hook unconditionally that was created in the outer scope. That whole statement looks really scary.

Reply:
Hum 🤔